test/spec/fixtures/schema.sql

```diff
@@ -2294,7 +2292,8 @@ A test for partitioned tables$$;
         car_dealer_city varchar(64) not null,
         quantity int not null,
         foreign key (car_model_name, car_model_year) references test.car_models (name, year),
-        foreign key (car_dealer_name, car_dealer_city) references test.car_dealers (name, city)
+        foreign key (car_dealer_name, car_dealer_city) references test.car_dealers (name, city),
+        primary key (car_model_name, car_model_year, car_dealer_name, car_dealer_city, quantity)
```

Comment:
I believe this new detection makes perfect sense. It also accounts for cases where the PK has an additional column(`quantity`) that is not part of an FK.